rotkehlchen/db/snapshots.py

```diff
@@ -223,55 +223,69 @@ def _export(
     ) -> Tuple[bool, str]:
-        """
-        Converts the snapshot files to list to dictionaries.
-        Performs a series of validation checks on the list before importing.
-        """
-        balances_list = self._csv_to_dict(balances_snapshot_file)
-        location_data_list = self._csv_to_dict(location_data_snapshot_file)
+        """Validates the snapshot data about to be imported."""
         # check if the headers match the type stored in the db
         has_invalid_headers = (
-            tuple(balances_list[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', 'usd_value') or  # noqa: E501
-            tuple(location_data_list[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
+            tuple(balances_data[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', 'usd_value') or  # noqa: E501
+            tuple(location_data[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
         )
         if has_invalid_headers:
-            return False, 'csv file has invalid headers'
+            return False, f'{source} has invalid headers'
 
+        # check if the timestamp can be converted to int
+        try:
+            balances_timestamps = [deserialize_timestamp(entry['timestamp']) for entry in balances_data]  # noqa: E501
+            location_data_timestamps = [deserialize_timestamp(entry['timestamp']) for entry in location_data]  # noqa: E501
+        except DeserializationError:
+            return False, f'{source} contains invalid timestamp format'
         # check if all timestamps are the same.
-        balances_timestamps = [int(entry['timestamp']) for entry in balances_list]
-        location_data_timestamps = [int(entry['timestamp']) for entry in location_data_list]
         has_different_timestamps = (
             balances_timestamps.count(balances_timestamps[0]) != len(balances_timestamps) or
             location_data_timestamps.count(location_data_timestamps[0]) != len(location_data_timestamps) or  # noqa: E501
             balances_timestamps[0] != location_data_timestamps[0]
         )
-        if has_different_timestamps:
-            return False, 'csv file has different timestamps'
+        if has_different_timestamps is True:
+            return False, f'{source} has different timestamps'
 
-        # check if the timestamp can be converted to int
-        try:
-            _ = deserialize_timestamp(balances_list[0]['timestamp'])
-        except DeserializationError:
-            return False, 'csv file contains invalid timestamp format'
+        return True, ''
 
-        return self._import_snapshot(
-            balances_list=balances_list,
-            location_data_list=location_data_list,
+    def parse_import_snapshot_data(
+            self,
+            balances_snapshot_file: Path,
+            location_data_snapshot_file: Path,
+    ) -> Tuple[bool, Dict[str, Any]]:
+        """Converts the snapshot files to list to dictionaries.
```

Comment:
The docstring makes no sense.

Describe what is returned. `List to dictionaries` is not a thing. I see you return a tuple. Tuple's first element is a bool. What is that bool supposed to be? What does the second element which is a dict supposed to contain? This is what the docstring should describe.